src/lib/featureFlagSlice.ts

```diff
@@ -0,0 +1,45 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+// const API_ADDRESS = "http://localhost:8080";
+
+export enum FFState {
+  Enabled = "Enabled",
+  Disabled = "Disabled",
+  Pending = "Pending",
+}
+
```

Comment:
Regarding typing here, I think we need some more discussions.

@AndrewMuh, do you think it's a better idea if we give each flag its own type for granular state mangament?
If so, should we implement it in union types? (given that enum cannot be extended) Or totally separate types?

Also, do you think the rendering logic like `senior-only` should be handled at backend or front-end?

This question is important, because it will affect how `FeatureFlagElement` should be implemented (in Nav.tsx)

Reply:
Honestly thinking about it now, I think all flags should be either Enabled or Disabled (I still think enums, but only these two). Things like senior only or student only should be done through a "featureIsSeniorOnly" feature flag. This simplifies implementation in the frontend/db: just check ephmatchEnabled and ephmatchIsSeniorOnly for example.